modules/plugin/epsg-hsql/src/test/java/org/geotools/referencing/PROJParsingTest.java

```diff
@@ -0,0 +1,134 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2024, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.referencing;
+
+import static org.junit.Assert.assertTrue;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Properties;
+import org.geotools.api.referencing.FactoryException;
+import org.geotools.api.referencing.crs.CoordinateReferenceSystem;
+import org.geotools.api.referencing.crs.GeographicCRS;
+import org.geotools.api.referencing.crs.ProjectedCRS;
+import org.geotools.referencing.proj.PROJFormattable;
+import org.geotools.referencing.proj.PROJFormatter;
+import org.geotools.test.TestData;
+import org.junit.Ignore;
+import org.junit.Test;
+
+public class PROJParsingTest {
+
+    /**
+     * Test used to create the definition files. Keep ignoring-it unless you need to generate the
+     * definition files again
+     */
+    @SuppressWarnings("PMD.SystemPrintln")
+    @Test
+    @Ignore
+    public void testGeneratePROJDefinitions() throws IOException {
```

Comment:
Is this actually needed? I believe we should have a list of proj strings generated from proj, and an integration test that compares them with versions generated by GeoTools.
The latter we do not need to dump, we just need to collect the differences and report them in output at the end of the test, when there is a failure, using "fail(message)" (no need to use System.out)

Reply:
The existing test (not the one being ignored) is creating PROJ definition for each EPSG code and it's comparing it with the expected projection created by GeoTools. Note that PROJ official String and GEOT-PROJ String may be different. 
The relevant part is that the PROJ library produces the same CRS for both when parsing it.
Since we don't have a PROJ parser in our tests, what we can do is to check whether the created GEOT-PROJ is matching a GEOT-PROJ String that is creating the same CRS as PROJ for that EPSG code.

The ignored test can be used to create the "expected" list (and the python gt_proj_compare.py script can be used to compare it with PROJ definitions)